File: backend/app/models/architecture.py
from enum import Enum
from pydantic import BaseModel


class Confidence(str, Enum):
    VERIFIED = "verified"
    INFERRED = "inferred"
    UNVERIFIABLE = "unverifiable"


class Component(BaseModel):
    id: str
    name: str
    role: str
    path: str
    confidence: Confidence
    children: list[str] = []


class Relationship(BaseModel):
    source_id: str
    target_id: str
    type: str
    confidence: Confidence


class Architecture(BaseModel):
    repo_url: str
    components: list[Component] = []
    relationships: list[Relationship] = []

File: frontend/src/types/architecture.ts
export type Confidence = 'verified' | 'inferred' | 'unverifiable'

export interface Component {
  id: string
  name: string
  role: string
  path: string
  confidence: Confidence
  children: string[]
}

export interface Relationship {
  source_id: string
  target_id: string
  type: string
  confidence: Confidence
}

export interface Architecture {
  repo_url: string
  components: Component[]
  relationships: Relationship[]
}
